xrpl/models/transactions/transaction.py

```diff
@@ -468,3 +468,43 @@ def from_blob(tx_blob: str) -> Transaction:
             The formatted transaction.
         """
         return Transaction.from_xrpl(decode(tx_blob))
+
+    @classmethod
+    def from_xrpl(cls: Type[T], value: Union[str, Dict[str, Any]]) -> T:
+        """
+        Creates a Transaction object based on a JSON or JSON-string representation of
+        data
+
+        In Payment transactions, the DeliverMax field is renamed to the Amount field.
+
+        Args:
+            value: The dictionary or JSON string to be instantiated.
+
+        Returns:
+            A Transaction object instantiated from the input.
+
+        Raises:
+            XRPLModelException: If Payment transactions have different values for
+                                amount and deliver_max fields
+        """
+        processed_value = process_xrpl_json(value)
+
+        # handle the deliver_max alias in Payment transactions
+        if (
+            "transaction_type" in processed_value
+            and processed_value["transaction_type"] == "Payment"
+        ):
+            if "deliver_max" in processed_value:
+                if "amount" in processed_value:
+                    if processed_value["amount"] != processed_value["deliver_max"]:
+                        raise XRPLModelException(
+                            "Error: Amount and DeliverMax fields are not identical"
```

Comment:
Is there maybe a link we can give to explain why they should be equal in the error message?

Reply:
Hmm, I can point to the issue discussion here: https://github.com/XRPLF/rippled/issues/3484

Here is an implementation commit to the rippled codebase: https://github.com/XRPLF/rippled/commit/397268394b80574419fddafe77aa75ea11ad9cfe 

I'm not aware of any other documents summarizing this issue.

`DeliverMax` is an alias for the `Amount` field. So if they contain different values, it introduces semantic ambiguity. Which one should we use? Both of them refer to the same logical quantity -- the value of transferred money

Reply:
If both fields are passed in, then they have to be equal. We can update the error message for clarity: `"Error: Amount and DeliverMax fields must be equal if both are provided"`

Reply:
Don't link rippled issues from the libraries, point to the xrpl.org documentation. It's not useful to do so in error messages, because that makes the error message really long, but it can be in the docstring.

Reply:
I was brainstorming ideas for resource pages for this issue, I won't include it in the error message 👍 

I couldn't find a statement that says "If present, `DeliverMax` field must equal `Amount`" in the Payments docs. https://xrpl.org/docs/references/protocol/transactions/types/payment#payment-fields 

I feel it's a better developer experience to provide an elaborate error message, instead of making them search through a website.

Reply:
`DeliverMax` is only an RPC field, rippled doesn't care about it. It's only ever returned. That documentation should probably elaborate on that a bit.

Reply:
Yes, documentation will help. `DeliverMax` can also be used as an input in APIv2 rippled. 

Every instance of `DeliverMax`, at the protocol (or) binary-codec-level, is translated into `Amount`.

Reply:
No, it was not changed at the protocol level, just at the RPC level.